frontend/admin/src/js/components/Layout.tsx

```diff
@@ -47,9 +48,20 @@ const OpenMenuButton: React.FC<OpenMenuButtonProps> = ({
 const Layout = () => {
   const intl = useIntl();
   const isSmallScreen = useIsSmallScreen();
-  const [isMenuOpen, setMenuOpen] = React.useState(!isSmallScreen);
+
+  // use local storage for menu preference, true = open and false = closed
+  const [menuPreference, setMenuPreference] = useLocalStorage(
+    "digitaltalent-menustate",
+    true,
```

Comment:
I don't think so, the initial value is only used if the key doesn't exist in storage right?
So if you have the key already, but lets say your browser was resized when you are visiting the page today, then your small screen won't be picked up

Reply:
doesn't seem to work right
I think it is due to this bit
```
  React.useEffect(() => {
    setMenuOpen(!isSmallScreen);
  }, [isSmallScreen, setMenuOpen]);
```
now even if menuOpen is false the page loads with menu opened for me
`!isSmallScreen` returns true and is probably the issue, overriding my menu preference
this may be complicated by Strict Mode being enabled so useEffect fires twice right? so I don't really want to throw a useEffect in here if not absolutely needed

Reply:
the suggestion however, inspired me to make it so the shrinking the browser window will collapse the menu automatically
added a different useEffect in https://github.com/GCTC-NTGC/gc-digital-talent/pull/5351/commits/2a917a52fbae339b5721bddfcedba250febff84f
think this one should be safe since firing twice or thrice makes no difference as it isn't a toggle, it only sets things to false

Reply:
not really seeing the conditional as something that troublesome

Reply:
What is the issue that occurs when you remove the `useState` hook and only rely on local storage? I can't seem to find any issues with it and right now the `useState` just seems to be an unneeded redundancy

Reply:
merging the ideas together seems to work fine! https://github.com/GCTC-NTGC/gc-digital-talent/pull/5351/commits/c27191f4c5748e08f512a2a9dd5ffeff6eb419cc
`setMenuOpen(!isSmallScreen);` was just the troublesome bit